### PR TITLE
remove temp variable

### PR DESCRIPTION
**Pull request by horance-liu**
1. I remove temp variable, and return it directly. 
2. using `auto` deduction.

### CODE REVIEW

File: tensorflow/core/distributed_runtime/rpc/grpc_worker_cache.cc [hunk 1]
```diff
@@ -43,7 +43,7 @@ class GrpcWorkerCache : public WorkerCachePartial {
           void* tag;
           bool ok;
           while (completion_queue_.Next(&tag, &ok)) {
-            GrpcClientCQTag* callback_tag = static_cast<GrpcClientCQTag*>(tag);
+            auto callback_tag = static_cast<GrpcClientCQTag*>(tag);
```
Comment (jhseu) [inline tensorflow/core/distributed_runtime/rpc/grpc_worker_cache.cc]: Mind excluding these auto changes? It doesn't match the majority of TF style.

Comment (horance-liu) [inline tensorflow/core/distributed_runtime/rpc/grpc_worker_cache.cc]: I have seen many examples for using `auto`, and here there is no need to write `GrpcClientCQTag*` with two times. So, I apply this change not only to use `auto` deduction, but to eliminate duplicate type declaration.

**Comment by horance-liu** [comment]
I have rollback modification, and please review again.

**Review by jhseu — APPROVED** [review]

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Pull request merged**